olp-cpp-sdk-dataservice-write/include/olp/dataservice/write/FlushEventListener.h

```diff
@@ -141,28 +89,28 @@ class DefaultFlushEventListener : public FlushEventListener<FlushResponse> {
    */
   void NotifyFlushEventResults(FlushResponse results) override;
 
+  void NotifyFlushMetricsHasChanged(FlushMetrics metrics) override{};
+
  protected:
   template <typename T>
   bool CollateFlushEventResults(const std::vector<T>& results) {
-    total_flush_requests_ += results.size();
+    metrics_.num_total_flushed_requests += results.size();
```

Comment:
need to lock?

Reply:
No, `CollateFlushEventResults` is called from method, which already holds a lock.